app/test_common.go

```diff
@@ -194,7 +196,7 @@ func (tApp TestApp) InitializeFromGenesisStatesWithTimeAndChainIDAndHeight(genTi
 // CheckBalance requires the account address has the expected amount of coins.
 func (tApp TestApp) CheckBalance(t *testing.T, ctx sdk.Context, owner sdk.AccAddress, expectedCoins sdk.Coins) {
 	coins := tApp.GetBankKeeper().GetAllBalances(ctx, owner)
-	require.Equal(t, expectedCoins, coins)
+	require.True(t, coins.IsEqual(expectedCoins))
```

Comment:
this func wasn't used anywhere so it felt safe to change.
i ran into a case where an empty coins array when converted to json and back gives `nil` instead of an empty coins slice.
the coins.IsEqual allows correct comparison with nil (both have zero value). 

either check works for the test i use this in, so i'll just revert it back 👍